### PR TITLE
Add acquisition hdr setters

### PR DESCRIPTION
**Pull request by johannesmayer**
## Changes in this pull request
This adds the ability to modify the Acquisitions Header Index information using SIRF.
This is useful (e.g.) in cardiac cine reconstruction if some phases are not fully sampled to add acquisitions from one phase to another prior to reconstruction such that all phases have fully sampled data points.
Also, since the header information can be modified, I also added the possibility to change the limits of encoding spaces in the header so that any discrepancy between headers of individual readouts and the size of the reconstructed space can be avoided.

There is no check if the set acquisition header information is compatible with the limits in the header.
This will potentially lead to a crash in reconstruction.

## Testing performed
Setter and getter of the limits are checked on C++ and Python (though in Python you need to check manually if the test prints the correct output).

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added docstrings/doxygen in line with the guidance in the developer guide
- [x] I have implemented unit tests that cover any new or modified functionality
- [x] The code builds and runs on my machine
- [ ] CHANGES.md has been updated with any functionality change

## Contribution Notes

Please read and adhere to the [contribution guidelines](https://github.com/SyneRBI/SIRF/blob/master/CONTRIBUTING.md).

Please tick the following: 

 - [x] The content of this Pull Request (the Contribution) is intentionally submitted for inclusion in SIRF (the Work) under the terms and conditions of the [Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0) License.

### CODE REVIEW

File: src/xGadgetron/cGadgetron/include/sirf/Gadgetron/cgadgetron.h [hunk 1]
```diff
@@ -58,6 +58,8 @@ extern "C" {
 	void* cGT_setAcquisitionModelParameter
 		(void* ptr_am, const char* name, const void* ptr);
 	void* cGT_AcquisitionModelParameter(void* ptr_am, const char* name);
+	void* cGT_setCSParameter(void* ptr, const char* par, const void* val);
+	void* cGT_setAcquisitionParameter(void* ptr, const char* param_name, const void* val);
```
Comment (evgueni-ovtchinnikov) [inline src/xGadgetron/cGadgetron/include/sirf/Gadgetron/cgadgetron.h:62]: not needed - Python interface calls `cGT_setParameter` (which calls one of these), see `Gadgetron_params.py`.

Comment (johannesmayer) [reply src/xGadgetron/cGadgetron/include/sirf/Gadgetron/cgadgetron.h]: Deleted the declarations

**Comment by KrisThielemans** [comment]
@johannesmayer @ckolbPTB is this ready?

There's an unrelated STIR build error with HDF5 errors causing a job failure unfortunately. Apologies for that.

**Comment by ckolbPTB** [comment]
Yes, this can be merged.

**Comment by KrisThielemans** [comment]
@evgueni-ovtchinnikov can you do the honours?

**Comment by evgueni-ovtchinnikov** [comment]
@johannesmayer @ckolbPTB could you please add a line or two describing what have been done in CHANGES.md?

**Pull request merged**